dictionary/MessageFormatDictionary.js

```diff
@@ -80,22 +80,54 @@ export default class MessageFormatDictionary extends Dictionary {
 	 *        Defaults to an empty plain object.
 	 */
 	get(key, parameters = {}) {
+		let scope = this._getScope(key);
+
+		if (scope === false) {
+			throw new GenericError(
+				`ima.dictionary.MessageFormatDictionary.get: The ` +
+				`localization phrase '${key}' does not exists`,
+				{ key, parameters }
+			);
+		}
+
+		return scope(parameters);
+	}
+
+	/**
+	 * @inheritdoc
+	 * @param {string} key The key identifying the localization phrase. The key
+	 *        consists of at least two parts separated by dots. The first part
+	 *        denotes the name of the source JSON localization file, while the
+	 *        rest denote a field path within the localization object within
+	 *        the given localization file.
+	 */
+	has(key) {
+		let scope = this._getScope(key);
+		return (scope === false) ? false : true;
+	}
+
+	/**
+	 * @inheritdoc
+	 * @param {string} key The key identifying the localization phrase. The key
+	 *        consists of at least two parts separated by dots. The first part
+	 *        denotes the name of the source JSON localization file, while the
+	 *        rest denote a field path within the localization object within
+	 *        the given localization file.
+	 */
+	_getScope(key) {
 		let path = key.split('.');
 		let scope = this._dictionary;
 
 		for (let scopeKey of path) {
 			if (!scope[scopeKey]) {
-				throw new GenericError(
-					`ima.dictionary.MessageFormatDictionary.get: The ` +
-					`localization phrase '${key}' does not exists`,
-					{ key, parameters }
-				);
+				return false;
 			}
 
 			scope = scope[scopeKey];
 		}
 
-		return scope(parameters);
+		// only if scope is function, the key is useable
+		return (typeof scope === "function") ? scope : false;
```

Comment:
This method is meant to retrieve an object (a sub-scope) or a localization phrase generator (a function), both of which are objects. The correct way of signalling a non-existence of an object is returning `null`. Therefore, please update this statement as follows:

`return scope;`